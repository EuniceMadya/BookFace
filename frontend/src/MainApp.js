
import React, { useState, useEffect } from 'react';

import { UserContext } from './UserContext';
import App from './App'

class MainApp extends React.Component {
  constructor(props) {
    super(props);

    this.toggleToken = (data) => {
      this.setState(state => ({
        token : data.token,
        userId: data.id,
      }));
    };

    this.setloggedIn = (value) => {
      this.setState(state => ({
        loggedIn: value
      }));
    }

    this.state = {
      token: "default token",
      toggleToken: this.toggleToken,
      loggedIn: false,
      userId: 0,
      setloggedIn: this.setloggedIn
    };
  }

  render() {
    return (
      <UserContext.Provider value={this.state}>
        <App />
      </UserContext.Provider>
    );
  }
}

export default MainApp;

