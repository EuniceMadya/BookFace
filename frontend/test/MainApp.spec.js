import MainApp from '../src/MainApp';
import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';


describe('MainApp', () => {
    it('renders without error', () => {
	  const view = renderer.create(MainApp).toJSON();
	  expect(view).toMatchSnapshot();
	});
});

