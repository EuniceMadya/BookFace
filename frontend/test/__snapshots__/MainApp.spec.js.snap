// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainApp renders without error 1`] = `null`;
